.github/workflows/ci.yml

```diff
@@ -19,62 +19,119 @@ jobs:
         include:
```

Comment:
(meta thread)

- I believe we should keep test+php-cs-fixer self-run together. The self-run us form of a test for us as well, (not only checking the style, but also checking if new code is not causing any crashes during execution, for path that was not detected with utests - consider this kind of big-level smoke tests)
- I extracted many good elements from this PR and merged them independently already

with that, I believe we can close the PR. Thanks!

Reply:
I don't agree. Tests and Fixer are totally not related and both are executed on the same, existing code. Nothing is changed during their execution and there's no need for them to be run one after another in the same job. I don't see a value in stacking these steps, leading to longer job that requires looking into logs to see which part failed. As far as I can agree on other points (like minimal deps etc), I just don't agree here.

Reply:
I don't enjoy the part where we need to set up PHP twice and then... we need to run migration rules twice - once for "utests", once for "fixer check" - ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7366#discussion_r1373387897

with that, I do not see big gain to split the job.

Reply:
pls address that part too
> but you see this split bringing missing elements (like the link i pointed)
> we need to run migration rules twice - once for "utests", once for "fixer check" - ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7366#discussion_r1373387897